src/sxg_codec.c

```diff
@@ -243,22 +227,49 @@ static const EVP_MD* select_digest_function(EVP_PKEY* key) {
   }
 }
 
-bool sxg_evp_sign(EVP_PKEY* private_key, const sxg_buffer_t* src,
-                  sxg_buffer_t* dst) {
-  EVP_MD_CTX* const ctx = EVP_MD_CTX_new();
+size_t sxg_evp_sign_size(EVP_PKEY* private_key, const uint8_t* src,
+                         size_t length) {
+  EVP_MD_CTX* const ctx = EVP_MD_CTX_create();
   const EVP_MD* digest_func = select_digest_function(private_key);
-
-  // EVP_PKEY_sign_init() and EVP_PKEY_sign() return 1 for success and 0 or a
-  // negative value for failure. In particular a return value of -2 indicates
-  // the operation is not supported by the public key algorithm.
   size_t sig_size = 0;
   bool success =
       ctx != NULL &&
       (EVP_DigestSignInit(ctx, NULL, digest_func, NULL, private_key) == 1) &&
-      (EVP_DigestSign(ctx, NULL, &sig_size, src->data, src->size) == 1) &&
-      sxg_buffer_resize(sig_size, dst) &&
-      (EVP_DigestSign(ctx, dst->data, &dst->size, src->data, src->size) == 1);
+      (EVP_DigestSign(ctx, NULL, &sig_size, src, length) == 1);
+  EVP_MD_CTX_destroy(ctx);
+  if (!success) {
+    return 0;
+  }
+  return sig_size;
+}
 
-  EVP_MD_CTX_free(ctx);
-  return success;
+size_t sxg_evp_sign(EVP_PKEY* private_key, const uint8_t* src, size_t length,
+                    uint8_t* dst) {
+  EVP_MD_CTX* const ctx = EVP_MD_CTX_create();
+  const EVP_MD* digest_func = select_digest_function(private_key);
+  // EVP_PKEY_sign_init() and EVP_PKEY_sign() return 1 for success and 0 or a
+  // negative value for failure. In particular a return value of -2 indicates
+  // the operation is not supported by the public key algorithm.
+  size_t sig_size = 72;
+  printf("evp signing with %p, %d, %p\n", private_key, length, dst);
+  bool success = ctx != NULL && (EVP_DigestSignInit(ctx, NULL, digest_func,
+                                                    NULL, private_key) == 1);
+  printf("evp init with %p, %p, %p :%d\n", ctx, digest_func, private_key,
+         success);
+  int result = EVP_DigestSign(ctx, dst, &sig_size, src, length);
+  success = success && (result == 1);
+  if (result != 1) {
+    char* buf = OPENSSL_malloc(4096);
+    for (size_t i = 0; i < 4096; ++i) {
+      buf[i] = '\0';
```

Comment:
`memset(buf, 0, 4096);`  also use constant for size.

Reply:
Sorry, this is debug purpose code. already deleted.